server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -2162,11 +2163,13 @@ private void deleteSnapshotsFromRepository(
             assert currentlyFinalizing.contains(deleteEntry.repository());
             final List<SnapshotId> snapshotIds = deleteEntry.getSnapshots();
             assert deleteEntry.state() == SnapshotDeletionsInProgress.State.STARTED : "incorrect state for entry [" + deleteEntry + "]";
+            RemoteStoreLockManagerFactory remoteStoreLockManagerFactory = new RemoteStoreLockManagerFactory(() -> repositoriesService);
             repositoriesService.repository(deleteEntry.repository())
                 .deleteSnapshots(
                     snapshotIds,
                     repositoryData.getGenId(),
                     minCompatibleVersion(minNodeVersion, repositoryData, snapshotIds),
+                    remoteStoreLockManagerFactory,
```

Comment:
wondering if there is a way to move `remoteStoreLockManagerFactory` to `BlobStoreRepository` instance instead of passing it in the API ?

Reply:
Invoking the cleanup separately before `deleteSnapshots` is called, leaving it's signature as it is.